sqlglot/transforms.py

```diff
@@ -324,18 +346,27 @@ def unnest_to_explode(expression: exp.Expression) -> exp.Expression:
             unnest = join_expr.this if is_lateral else join_expr
 
             if isinstance(unnest, exp.Unnest):
-                alias = join_expr.args.get("alias") if is_lateral else unnest.args.get("alias")
-                udtf = exp.Posexplode if unnest.args.get("offset") else exp.Explode
+                alias = (
+                    join_expr.args.get("alias")
+                    if is_lateral and join_expr.args.get("alias")
+                    else unnest.args.get("alias")
```

Comment:
Why was the new condition (RHS of the `and`) needed here?

Also, ditto about the `arg` lookups

Reply:
We set the default alias in `starrocks.py`:
```
unnest.set("alias", alias)
```
The following `starrocks sql` will use this logic:
```
SELECT student, score, unnest FROM tests CROSS JOIN LATERAL UNNEST(scores)
```
But I found that ` join_expr.args.get("alias")` may not get alias, it is only in Unnest, `sqlglot.dialects.starrocks.StarRocks.Parser._parse_unnest` cannot set the lateral alias.

I think when is_lateral, it cannot be obtained from outside, and the default value we set should be obtained from unnest.args. Am I missing something?

I provide details:

```
Select(
  expressions=[
    Column(
      this=Identifier(this=student, quoted=False)),
    Column(
      this=Identifier(this=score, quoted=False)),
    Column(
      this=Identifier(this=unnest, quoted=False))],
  from=From(
    this=Table(
      this=Identifier(this=tests, quoted=False))),
  joins=[
    Join(
      this=Lateral(
        this=Unnest(
          expressions=[
            Column(
              this=Identifier(this=scores, quoted=False))],
          alias=TableAlias(
            columns=[
              Identifier(this=unnest, quoted=False)]))),
      kind=CROSS)])
```